### ISSUE TITLE
Tasks 2.0 npm run compile task as a preLaunchTask, Code won't start debugger

### ISSUE THREAD

**Issue by ytimenkov** [Microsoft/vscode#29014]
- VSCode Version: Code 1.13.1 (379d2efb5539b09112c793d3d9a413017d736f89, 2017-06-14T18:21:47.485Z) 
- tasks.json:
~~~json
{
    "version": "2.0.0"
}
~~~
- launch.json
~~~json
// A launch configuration that compiles the extension and then opens it inside a new window
{
    "version": "0.1.0",
    "configurations": [
        {
            "name": "Launch Extension",
            "type": "extensionHost",
            "request": "launch",
            "runtimeExecutable": "${execPath}",
            "args": ["--extensionDevelopmentPath=${workspaceRoot}" ],
            "stopOnEntry": false,
            "sourceMaps": true,
            "outFiles": [
                "${workspaceRoot}/out/*",
                "${workspaceRoot}/out/src/*",
                "${workspaceRoot}/out/test/*"
            ],
            "preLaunchTask": "run compile",
            "smartStep": true
        }
    ]
}
~~~

- *Expected result*: When I press F5, VS Code runs npm run compile and then starts debugger.
- *Actual result*:
    - When I press F5 first time VS Code just compiles and doesn't start debugger, just showing a spinning line in the status bar.
    - Subsequent attempts to run actually starts debugger and work as expected.
    - When I execute "Terminate running task" command I get error message:
    ![image](https://user-images.githubusercontent.com/7413219/27279663-a103961a-54e5-11e7-8a42-f06364bbe631.png)

Related to: #28235

**Comment by dbaeumer** [comment]
@ytimenkov where does the run compile task come from. Can you execute that task from F1 > Run Task ?

**Comment by dbaeumer** [comment]
Info is in #29013

**Comment by ytimenkov** [comment]
@dbaeumer This task is provided by `npm` built-in extension, activated when `tasks.json` has version 2.0.0.
I suspect there are either some glitches in handling `isBackground` in new tasks implementation (or it is not supported or implemented in npm extension) or `tsc watch` requires special parser which lets vscode know when initial job is done and task went into watch mode.

**Comment by dbaeumer** [comment]
This for addressed with the fix done for #29013. Marking as dup: #29013
